### ISSUE TITLE
After Fixing coupon Code Issue Facing New Issue Updating Price from Admin Side

### ISSUE THREAD

**Issue by mrabdulhannan** [magento/magento2#37872]
### Preconditions and environment

Magento Version: 2.4.6
Additional Information for Reproduction: If you're looking into this issue, please note that I faced a problem with applying a coupon code multiple times, even though I had set restrictions for it to be used only once. I referred to the solution provided in the following link to resolve the issue:
[Adobe Commerce Knowledge Base - Coupon Code Used More Than Once](https://experienceleague.adobe.com/docs/commerce-knowledge-base/kb/troubleshooting/miscellaneous/coupon-code-used-more-than-once-adobe-commerce.html?lang=en).
As part of the solution, I inserted the following code into my env.php file:
php
Copy code
'cron_consumers_runner' =>
    [
        'cron_run' => true,
        'max_messages' => 20000,
        'consumers' =>
        [
            'sales.rule.update.coupon.usage'
        ]
    ]
This successfully resolved the coupon code issue. However, it gave rise to a new problem related to updating product prices from the admin side.
Please let me know if you need any further details or if you have any suggestions on how to address this new issue.

### Steps to reproduce

Certainly, here's a more descriptive rephrasing of the steps to reproduce the issue:

**Create a Configurable Product:**
Begin by generating a Configurable product within the Magento instance. This involves setting up the product's attributes and variations to provide multiple options for customers.

**Update Price from Admin Side:**
Navigate to the admin dashboard and access the product management section. In the process, locate the Configurable product you created. Employ the action column to manually adjust the price of the product.

**Unresponsive Price Update:**
Regrettably, upon changing the price through the admin interface, the anticipated automatic update of the price does not occur as expected for the Configurable product.

Execute php bin/magento cron:run Command:
In an effort to rectify the non-updating price issue, execute the command php bin/magento cron:run from the command line interface. The execution of this command results in the successful update of the product's price.

**Issue Resurfacing After Multiple Runs:**
As a perplexing twist, after repeatedly running the php bin/magento cron:run command several times, the issue of the price not updating resurfaces once again.

### Expected result

**Unyielding Price Update Issue:**
Despite resorting to the command-line execution mentioned above, the problem persists, and the price of the Configurable product remains unaltered.

**Exhaustive Attempts to Resolve:**
In an attempt to troubleshoot and address the issue comprehensively, you have diligently executed a series of pertinent commands. Unfortunately, even after executing these commands, the problem persists, and the expected price update remains elusive.

Should you require additional insights, guidance, or further assistance in unraveling this perplexing issue, please do not hesitate to communicate.

### Actual result

Price Is not getting changed from the admin side

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [X] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by mrabdulhannan** [comment]
@magento give me 2.4.6-develop instance

**Comment by engcom-Bravo** [comment]
Hi @mrabdulhannan,

Thank you for reporting and collaboration.

Verified the issue on Magento 2.4-develop instance and the issue is not reproducible.Kindly refer the screenshots.

**We have created Cart price rule** :

![Test-Promotions-Marketing-Magento-Admin (3)](https://github.com/magento/magento2/assets/51680745/365876de-3c96-479d-8ea2-362107a00e3f)



We have updated coupon usage with this command `bin/magento queue:consumers:start sales.rule.update.coupon.usage` as per this document https://experienceleague.adobe.com/docs/commerce-operations/configuration-guide/cli/start-message-queues.html. 


After updating coupon usage:

- We are able to **Create a Configurable Product**:
- We are able to **Update Price from Admin Side**: using **bin/magento cron:run** prices updated successfully.


The price also reflected in frontend also.


<img width="1480" alt="Screenshot 2023-08-16 at 3 27 16 PM" src="https://github.com/magento/magento2/assets/51680745/1d0212e4-fa64-4eba-965c-f275b860c636">


Kindly recheck the behaviour on Magento 2.4-develop instance and elaborate steps to reproduce if the issue is still reproducible.

Thanks.

**Comment by engcom-Bravo** [comment]
Hi @mrabdulhannan,

Kindly provide latest update on this comment https://github.com/magento/magento2/issues/37872#issuecomment-1680537789 still you are working on this issue if you are able to reproduce the issue please elaborate the steps to reproduce.Please let us know still if you are facing any issue.

Thanks.

**Comment by mrabdulhannan** [comment]
I installed the Magento 246-p2. This version does not have this issue on my end.

**Comment by engcom-Bravo** [comment]
Hi @mrabdulhannan,

Thanks for your update.

As per your comment https://github.com/magento/magento2/issues/37872#issuecomment-1697441085 so we are closing it. Please feel to raise a fresh ticket or reopen this ticket if you need more assistance on this.
Thanks.
